tsdb/chunkenc/chunk_test.go

```diff
@@ -217,3 +217,40 @@ func benchmarkAppender(b *testing.B, newChunk func() Chunk) {
 
 	fmt.Println("num", b.N, "created chunks", len(chunks))
 }
+
+func TestXor(t *testing.T) {
+	timestamp := int64(1234123324)
+	v := 83665193758603.123
+	num := 10
+	var samples []pair
+	for i := 0; i < num; i++ {
+		timestamp += int64(1000)
+		v += float64(100)
+		samples = append(samples, pair{t: timestamp, v: v})
+	}
+
+	chunk := NewXORChunk()
+	a, err := chunk.Appender()
+	testutil.Ok(t, err)
+	for _, p := range samples {
+		a.Append(p.t, p.v)
+	}
+
+	res := make([]pair, num)
+	var it Iterator
+	it = chunk.Iterator(it)
+	for it.Next() {
+		t, v := it.At()
+		res = append(res, pair{t: t, v: v})
+	}
+	if it.Err() != io.EOF {
+		testutil.Ok(t, it.Err())
+	}
```

Comment:
```suggestion
	testutil.ErrorEqual(t, io.EOF, it.Err())
```

Reply:
![image](https://user-images.githubusercontent.com/9583245/75446639-bd140000-59a2-11ea-8056-65839bdc58e5.png)

Reply:
Then that would be testutil.Ok() .. don't we always get nil ?

Reply:
In wchich case do we get EOF? is that `if` needed ?

Reply:
The code can't be changed like this. It can only be written in the original way

Reply:
So it means that sometimes it is EOF and other times nil ?

Reply:
![image](https://user-images.githubusercontent.com/9583245/75447541-8c34ca80-59a4-11ea-9cd7-c4abb1a5a736.png)
The "benchmark iterator" function also deals with this err problem

Reply:
funny, thanks